ws/config.py

```diff
@@ -1,180 +1,232 @@
-    if isinstance(value, bool):
-        return value
-    string = str(value).lower()
-    if string in ["1", "true", "yes"]:
+        self.configfile = configfile
+
+    def fetch_section(self, section=None, to_list=True):
+        """
+        Fetches a specific section from a config file.
+
+        :param section: section name for fetching.
+        :param to_list: defines the format of the returned value (see below).
+        :returns:
+            The data from the fetched config file section. The format is
+            a list of strings if ``to_list=True`` (default value) and
+            a dictionary otherwise.
+        """
+        with open(self.configfile) as f:
+            self.read_file(f)
+
+        if section is None:
+            section, _ = os.path.splitext(os.path.basename(sys.argv[0]))
+        if not self.has_section(section):
+            section = configparser.DEFAULTSECT
+
+        option_dict = dict(self.items(section))
+        for key, value in option_dict.items():
+            if ' ' in value:
+                option_dict[key] = value.split()
+        if to_list:
+            option_list = []
+            for key, value in option_dict.items():
+                option_list.append(key)
+                if isinstance(value, list):
+                    option_list.extend(value)
+                else:
+                    option_list.append(value)
+            
+        return option_list if to_list else option_dict
+
+
+class _ArgumentParser(argparse.ArgumentParser):
+    """Drop-in replacement for :py:class:`argparse.ArgumentParser`."""
+
+    def __init__(self, **kwargs):
+        kwargs.setdefault('usage', '%(prog)s [options]')
+        kwargs.setdefault('formatter_class',
+                          argparse.RawDescriptionHelpFormatter)
+        super().__init__(**kwargs)
+        # register new actions
+        self.register('action', 'check_path', _ActionCheckPath)
+        self.register('action', 'check_dirname', _ActionCheckDirname)
+
+    def setup(self):
+        """Initial argparser setup."""
+        
+        self.add_argument('-c', '--config',
+                          type=argtype_configfile,
+                          metavar='NAME',
+                          default=DEFAULT_CONF,
+                          help='name of config file (default: %(default)s')
+        self.add_argument('--cache-dir',
+                          type=argtype_dirname_must_exist,
+                          metavar='PATH',
+                          help=('directory for storing cached data'
+                                ' (will be created if necessary,'
+                                ' but parent directory must exist)'
+                                ' (default: %(default)s')
+                          
+        # some argument defaults that cannot be set with the 'default='
+        # parameter
+        arg_defaults = {}
+
+        cache_dir = os.getenv('XDG_CACHE_HOME', os.path.expanduser(CACHE_HOME))
```

Comment:
`os.path.expanduser` should be used when `CACHE_HOME` is defined.

Reply:
This was a typo - `CACHE_DIR` (not `CACHE_HOME`) is defined in the start of file: